homeassistant/components/vallox/sensor.py

```diff
@@ -21,170 +27,23 @@
-            unit_of_measurement=PERCENTAGE,
-            icon=None,
-            state_class=STATE_CLASS_MEASUREMENT,
-        ),
-        ValloxFilterRemainingSensor(
-            name=f"{name} Remaining Time For Filter",
-            state_proxy=state_proxy,
-            metric_key="A_CYC_REMAINING_TIME_FOR_FILTER",
-            device_class=DEVICE_CLASS_TIMESTAMP,
-            unit_of_measurement=None,
-            icon="mdi:filter",
-        ),
-        ValloxSensor(
-            name=f"{name} Efficiency",
-            state_proxy=state_proxy,
-            metric_key="A_CYC_EXTRACT_EFFICIENCY",
-            device_class=None,
-            unit_of_measurement=PERCENTAGE,
-            icon="mdi:gauge",
-            state_class=STATE_CLASS_MEASUREMENT,
-        ),
-        ValloxSensor(
-            name=f"{name} CO2",
-            state_proxy=state_proxy,
-            metric_key="A_CYC_CO2_VALUE",
-            device_class=DEVICE_CLASS_CO2,
-            unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
-            icon=None,
-            state_class=STATE_CLASS_MEASUREMENT,
-        ),
-    ]
-
-    async_add_entities(sensors, update_before_add=False)
-
-
 class ValloxSensor(SensorEntity):
     """Representation of a Vallox sensor."""
 
+    _attr_should_poll = False
+
     def __init__(
-        self,
-        name,
-        state_proxy,
-        metric_key,
-        device_class,
-        unit_of_measurement,
-        icon,
-        state_class=None,
+        self, name, state_proxy, description: ValloxSensorEntityDescription
     ) -> None:
         """Initialize the Vallox sensor."""
-        self._name = name
         self._state_proxy = state_proxy
-        self._metric_key = metric_key
-        self._device_class = device_class
-        self._state_class = state_class
-        self._unit_of_measurement = unit_of_measurement
-        self._icon = icon
-        self._available = None
-        self._state = None
-
-    @property
-    def should_poll(self):
-        """Do not poll the device."""
-        return False
-
-    @property
-    def name(self):
-        """Return the name."""
-        return self._name
-
-    @property
-    def native_unit_of_measurement(self):
-        """Return the unit of measurement."""
-        return self._unit_of_measurement
-
-    @property
-    def device_class(self):
-        """Return the device class."""
-        return self._device_class
-
-    @property
-    def state_class(self):
-        """Return the state class."""
-        return self._state_class
-
-    @property
-    def icon(self):
-        """Return the icon."""
-        return self._icon
-
-    @property
-    def available(self):
-        """Return true when state is known."""
-        return self._available
-
-    @property
-    def native_value(self):
-        """Return the state."""
-        return self._state
+
+        self.entity_description = description
+
+        # Entity Properties
```

Comment:
is this needed?

Reply:
Are you referring to the comment? It should go away?

Reply:
yes

Reply:
Removed it.